### DISCUSSION TITLE
example/tutorial on how to add new type of peripheral

### DISCUSSION THREAD

**Discussion by michcfr** [arendst/Tasmota#20819] [Q&A]
Hello,
I have searched for tutorials explaining on how to create a new type of peripheral, but did not found it. Do you know where I can find such tutorial? or any discussions explaining the steps? or any related example?

What I plan to do:
-integrate into Tasmtota a UART communication module that already works under Arduino.
-bridge my communication module with MQTT
-add a custom pages to WebUI to parameterize my module.

Thank you

**Comment by HofeBY** [comment]
https://github.com/phpcoder/tasmota-custom-integration
